src/ping.c

```diff
@@ -243,20 +283,20 @@ static void *ping_thread(void *arg) /* {{{ */
     return (void *)-1;
   }
 
-  if (ping_af != PING_DEF_AF) {
-    if (ping_setopt(pingobj, PING_OPT_AF, &ping_af) != 0)
+  if (sl->ping_af != PING_DEF_AF) {
+    if (ping_setopt(pingobj, PING_OPT_AF, &sl->ping_af) != 0)
       ERROR("ping plugin: Failed to set address family: %s",
             ping_get_error(pingobj));
   }
 
-  if (ping_source != NULL)
-    if (ping_setopt(pingobj, PING_OPT_SOURCE, (void *)ping_source) != 0)
+  if (sl->source != NULL && !sl->is_device)
+    if (ping_setopt(pingobj, PING_OPT_SOURCE, (void *)sl->source) != 0)
       ERROR("ping plugin: Failed to set source address: %s",
             ping_get_error(pingobj));
 
 #ifdef HAVE_OPING_1_3
-  if (ping_device != NULL)
-    if (ping_setopt(pingobj, PING_OPT_DEVICE, (void *)ping_device) != 0)
+  if (sl->source != NULL && sl->is_device)
+    if (ping_setopt(pingobj, PING_OPT_DEVICE, (void *)sl->source) != 0)
       ERROR("ping plugin: Failed to set device: %s", ping_get_error(pingobj));
 #endif
```

Comment:
It was previously possible to set *source address* **and** *device*. By using `sourcelist_t.source` for both, we lose this functionality.

I suggest that we have two `char *` fields in `sourcelist_t`, `src_addr` and `device`, and that we remove the `is_device` boolean. WDYT?